sources/Algo/include/Algo.h

```diff
@@ -323,20 +323,49 @@ struct HvdcLineDefinition {
   const ConverterId converter2_id;                             ///< second converter id
   const BusId converter2_busId;                                ///< second converter bus id
   const boost::optional<bool> converter2_voltageRegulationOn;  ///< second converter voltage regulation parameter, for VSC converters only
-  const Position position;                                     ///< position of the converters of this hvdc line compared to the main connex component
+  mutable Position position;                                   ///< position of the converters of this hvdc line compared to the main connex component
 };
 
+/**
+   * @brief Hash structure for term
+   *
+   * required to use set on terms
+   */
+struct HashTerm {
+  /**
+   * @brief Hash function
+   *
+   * @param hvdcLine the term to hash
+   * @returns the hash value of @p hvdcLine
+   */
+  std::size_t operator()(const HvdcLineDefinition& hvdcLine) const {
+    return std::hash<std::string>{}(hvdcLine.id);
+  }
+  /**
+   * @brief Less function
+   *
+   * @param lhs the left hand side to compare
+   * @param rhs the right hand side to compare
+   * @returns true if lhs is inferior to rhs based on their id
+   */
+  bool operator()(const HvdcLineDefinition& lhs, const HvdcLineDefinition& rhs) const {
+    return lhs.id < rhs.id;
+  }
+};
 /**
  * @brief the controller interface definition algorithm
  */
 class ControllerInterfaceDefinitionAlgorithm : public NodeAlgorithm {
  public:
+  using HvdcLineId = std::string;                              ///< HvdcLine id definition
+  using HvdcLineSet = std::set<HvdcLineDefinition, HashTerm>;  ///< TODO
```

Comment:
Do we need it to be sorted? Moreover a unordered map id->Definition might be better to avoid having to explicitly define operators?

Reply:
The issue I encountered when trying to make it unsorted is in the tests. With an unsorted container the order of the elements in a for loop is not always the same(I think). So if we write in the dyd file an hvdc before a lcc converter we will have an issue in the tests when we verify that the files are equal.

Reply:
ok, but we can still use a map to avoid heavy structures

Reply:
I don't get what will we gain from using a map ? Clarity of code ?

Reply:
yes

Reply:
Also the problem of set is that we cannot modify the key inplace